pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -54,230 +51,114 @@ func (e transientError) Error() string {
-				}
+		// Reconcile existing deployment replica counts which could have diverged
+		// outside the deployment process (e.g. due to auto or manual scaling).
+		// The active deployment is the last successful deployment, not
+		// necessarily the latest in terms of the config version. If the latest
+		// version failed to deploy, it should be scaled to zero and the last
+		// successful should be scaled back up.
+		activeDeployment := deployutil.ActiveDeployment(config, existingDeployments)
+		for _, deployment := range existingDeployments.Items {
+			oldReplicaCount := deployment.Spec.Replicas
+			newReplicaCount := oldReplicaCount
+			// The active deployment should be scaled to the config replica count, and
+			// everything else should be at zero.
+			if activeDeployment != nil && deployment.Name == activeDeployment.Name {
+				newReplicaCount = config.Template.ControllerTemplate.Replicas
+			} else {
+				newReplicaCount = 0
 			}
-			// Try to see if the image stream exists, if not then the build will never be able to update the
-			// tag in question
-			if _, err := c.osClient.ImageStreams(config.Namespace).Get(name); err != nil {
-				glog.V(2).Infof("Error while trying to get image stream %q: %v", name, err)
-				if errors.IsNotFound(err) {
-					details = fmt.Sprintf("The image trigger for image stream tag %q will have no effect because image stream %q does not exist.", istag, name)
+			// Only process updates if the replica count actually changed.
+			if oldReplicaCount != newReplicaCount {
+				deployment.Spec.Replicas = newReplicaCount
+				_, err := c.kubeClient.ReplicationControllers(deployment.Namespace).Update(&deployment)
+				if err != nil {
+					c.recorder.Eventf(config, "DeploymentScaleFailed", "Failed to scale deployment %q from %d to %d: %s", deployment.Name, oldReplicaCount, newReplicaCount, err)
+					return err
 				}
+				c.recorder.Eventf(config, "DeploymentScaled", "Scaled deployment %q from %d to %d", deployment.Name, oldReplicaCount, newReplicaCount)
 			}
 		}
-		allDetails = append(allDetails, details)
-	}
-
-	if len(allDetails) > 1 {
-		for i := range allDetails {
-			allDetails[i] = fmt.Sprintf("\t* %s", allDetails[i])
-		}
-		// Prepend multiple errors warning
-		multipleErrWarning := fmt.Sprintf("Deployment config %q blocked by multiple errors:\n", config.Name)
-		allDetails = append([]string{multipleErrWarning}, allDetails...)
+		return nil
 	}
-
-	return strings.Join(allDetails, "\n"), existingDeployments, latestDeploymentExists, nil
-}
-
-// updateDetails updates a deployment configuration with the provided details
-func (c *DeploymentConfigController) updateDetails(config *deployapi.DeploymentConfig, details string) (*deployapi.DeploymentConfig, error) {
-	if config.Details == nil {
-		config.Details = new(deployapi.DeploymentDetails)
+	// No deployments are running and the latest deployment doesn't exist, so
+	// create the new deployment.
+	deployment, err := deployutil.MakeDeployment(config, c.codec)
```

Comment:
This utility doesn't propagate the config namespace to the deployment resulting in errors when the controller is trying to create a new deployment.